src/ccall.cpp

```diff
@@ -692,10 +692,18 @@ static jl_cgval_t emit_cglobal(jl_codectx_t &ctx, jl_value_t **args, size_t narg
         }
         else {
             void *symaddr;
-            if (!jl_dlsym(jl_get_library(sym.f_lib), sym.f_name, &symaddr, 0)) {
+
+            void* libsym = jl_get_library_(sym.f_lib, 0);
+            if (!libsym || !jl_dlsym(libsym, sym.f_name, &symaddr, 0)) {
                 std::stringstream msg;
-                msg << "cglobal: could not find symbol ";
-                msg << sym.f_name;
+                if (!libsym) {
+                    msg << "cglobal: could not load library \n";
+                    // msg << jl_dlerror();
```

Comment:
@vtjnash any idea on how to make `jl_dlerror` work here?

Reply:
Repeat the lookup (fallback to the imaging mode code)